build-tools/automation/azure-pipelines.yaml

```diff
@@ -71,31 +98,41 @@ stages:
 
     # Prepare and build everything
     - script: make prepare-update-mono V=1 CONFIGURATION=$(XA.Build.Configuration) PREPARE_CI=1 PREPARE_AUTOPROVISION=1
+      workingDirectory: $(System.DefaultWorkingDirectory)/xamarin-android
```

Comment:
This part confuses me; how/why would `$(System.DefaultWorkingDirectory)/xamarin-android` exist? We have files such as `/Users/vsts/agent/2.155.1/work/1/s/bin/TestRelease/temp/SetupDependenciesForDesigner/UnnamedProject/foo.targets` (see e.g. cc21afa7ad222db77bcb4ce9f7484be40809aa08), i.e. `/Users/vsts/agent/2.155.1/work/1/s` is the clone of the xamarin-android directory.  Why would there be a `xamarin-android` directory *anywhere*?

Reply:
...or is one of the changes in this PR a change to that semantic?  In which case what *is* the `/Users/vsts/agent/2.155.1/work/1/s` directory?

Reply:
`$(System.DefaultWorkingDirectory)` is the clone of `xamarin-android` only if one `checkout` is specified. When multiple checkouts are specified, this default behavior changes, though we have some control over it. https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/multi-repo-checkout?view=azure-devops#checkout-path has some more info in that regard.